src/lexer.rs

```diff
@@ -34,384 +55,756 @@ pub fn lex(input: &str, allow_trailing_operators: bool, default_degree: Unit) ->
     }
   }
 
-  let mut chars = input.chars().peekable();
+  let graphemes = UnicodeSegmentation::graphemes(input.as_str(), true).collect::<Vec<&str>>();
```

Comment:
Parsing the graphemes makes sense, definitely easier than the wonkiness I was doing.

If I were to pass `$2000000000...`, currently cpc would read `$` and then return an error. With this, I think the lexer goes through the whole string first to parse the graphemes, and then starts reading the `$` to return the error.